### PR TITLE
Increase avocado timeouts

### PR DESCRIPTION
**Pull request by ldoktor**
This patch set increases some runner timeouts, gdb init timeout and improves the loop condition.

### CODE REVIEW

**Comment by ldoktor** [comment]
[BUG: Increase selftest timeouts (and related code)](https://trello.com/c/kftTQkwE/872-bug-increase-selftest-timeouts-and-related-code)

File: avocado/core/runner.py [hunk 1]
```diff
@@ -239,7 +239,8 @@ def finish(self, proc, started, timeout, step):
                        test_state['name'])
         if proc.is_alive():
             TEST_LOG.warning("Killing hanged test process %s" % proc.pid)
-            for _ in xrange(5):     # I really want to destroy it
+            end_time = time.time() + 60
+            while time.time() < end_time:
```
Comment (apahim) [inline avocado/core/runner.py]: Wow this is way more aggressive. We are going from 5 `os.kill()` calls in 0.5s to ~600 calls in 60s. Do we need that much of calls or all we need is to wait for one call to have effect?

File: avocado/core/runner.py [hunk 2]
```diff
@@ -366,7 +367,7 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
         time_started = time.time()
         proc.start()
 
-        test_status.wait_for_early_status(proc, 10)
+        test_status.wait_for_early_status(proc, 60)
```
Comment (apahim) [inline avocado/core/runner.py]: ok

File: avocado/utils/gdb.py [hunk 3]
```diff
@@ -675,10 +675,10 @@ def __init__(self, path='/usr/bin/gdbserver', port=None,
             self._wait_until_running()
 
     def _wait_until_running(self):
-        init_time = time.time()
         connection_ok = False
         c = GDB()
-        while time.time() - init_time < self.INIT_TIMEOUT:
+        end_time = time.time() + self.INIT_TIMEOUT
+        while time.time() < end_time:
```
Comment (apahim) [inline avocado/utils/gdb.py]: ok

File: avocado/core/runner.py (hunk 1 above)
Comment (ldoktor) [inline avocado/core/runner.py]: Honestly I don't know. SIGKILL is not maskable, so it should always be delivered and handled, so probably one call would suffice. On the other hand if there is a bug where the signal is not delivered then we might want to re-send it again.

The cost of re-sending the signal should be minimal as this signal can't be treated individually, so I'd prefer keeping it, but I'm open to execute it using `%10` or so. What do you think?

Comment (ldoktor) [inline avocado/core/runner.py]: PS: I know only of one example where killing takes longer than 5s and that is removing and re-creating a big file on `xfs` and there just one kill suffices, still it takes up to several minutes to finish...

Comment (apahim) [inline avocado/core/runner.py]: Ok, that (%10) will keep the number of calls around the same so we only touch the timeout here, which is the PR intention. Sounds good.

**Pull request closed without merging**